staging/src/k8s.io/apiserver/pkg/registry/rest/update.go

```diff
@@ -110,7 +110,12 @@ func BeforeUpdate(strategy RESTUpdateStrategy, ctx genericapirequest.Context, ob
 		return errors.NewInternalError(err)
 	}
 
-	errs = append(errs, strategy.ValidateUpdate(ctx, obj, old)...)
+	oldInitializers := oldMeta.GetInitializers()
+	if oldInitializers != nil && len(oldInitializers.Pending) != 0 {
+		errs = append(errs, createStrategy.Validate(ctx, obj)...)
```

Comment:
Has objectmeta mutation already been validated above?  If not, need to specifically call objectmeta mutation validation here (which is acceptable because we do that already for safety)

Reply:
Yes, it is. I think we need a method on update strategy for validating uninitialized objects that we, in *most* cases, delegate to Validate.  There are a very few objects, like services and their allocated IP addresses, that we still cannot allow to be mutated

Reply:
@liggitt how about making an optional `UpdateUninitializedStrategy` interface, and here we test if the updateStrategy implements the `UpdateUninitializedStrategy`, if not, delegate to createStrategy.Validate?

Reply:
> There are a very few objects, like services and their allocated IP addresses, that we still cannot allow to be mutated

By "allocated IP addresses", do you mean "ClusterIP"? Why can't it be mutated when the service is not initialized yet?

Reply:
Optional interfaces here are a recipe for unconsidered validation behavior (similar to GC defaults on the delete strategy). My preference would be to make it explicit

Reply:
So we need force people to consider if their resource needs special treatment when updating the uninitialized. How about adding back the "Mechanically adding ValidateUpdateUninitialized to all strategies" commit but delegating to `Validate()` instead of `ValidateUpdate()`?

Reply:
I think giving the strategy control via a ValidateUpdateUninitialized method is best... I'd recommend sweeping the ValidateUpdate functions while adding to see if there are things that initializers should not be allowed to change (like allocated service IPs)

also, there some registries (services, at least) do not use the generic store and would need special handling for uninitialized objects